nuxeo-opencmis-impl/src/main/java/org/nuxeo/ecm/core/opencmis/impl/server/NuxeoContentStream.java

```diff
@@ -114,6 +119,28 @@ public static boolean isHeadRequest(HttpServletRequest request) {
         return request.getMethod().equals("HEAD");
     }
 
+    public static boolean hasWantDigestRequestHeader(HttpServletRequest request, String digestAlgorithm) {
+        if (request == null || digestAlgorithm == null) {
+            return false;
+        }
+        if (request instanceof HttpServletRequestWrapper) {
+            request = (HttpServletRequest) ((HttpServletRequestWrapper) request).getRequest();
+        }
+        Enumeration<String> values = request.getHeaders("want-digest");
+        if (values == null) {
+            return false;
+        }
+        String digestAlgorithmUC = digestAlgorithm.toUpperCase();
+        Iterator it = Iterators.forEnumeration(values);
+        while (it.hasNext()) {
+            String value = ((String) it.next()).toUpperCase();
+            if (value.startsWith(digestAlgorithmUC)) {
```

Comment:
Why `startswith` and not straightforward `equalsIgnoreCase` comparison? If it's about the `;q=...` then it's better to split on `;` and use the first part for the actual digest.